src/mudlet.cpp

```diff
@@ -2399,35 +2399,40 @@ void mudlet::slot_replaySpeedDown()
 
 void mudlet::stopSounds()
 {
-    mpMusicBox1->stop();
-    mpMusicBox2->stop();
-    mpMusicBox3->stop();
-    mpMusicBox4->stop();
-}
-
-void mudlet::playSound( QString s )
-{
-    if( mpMusicBox1->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox1->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox1->play();
-    }
-    else if( mpMusicBox2->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox2->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox2->play();
-    }
-    else if( mpMusicBox3->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox3->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox3->play();
-    }
-    else
-    {
-        mpMusicBox4->stop();
-        mpMusicBox4->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox4->play();
-    }
+	QListIterator<QMediaPlayer *> itMusicBox( MusicBoxList );
+
+	while( itMusicBox.hasNext() ) {
+		itMusicBox.next()->stop();
+	} 
+}
+
+void mudlet::playSound( QString s, int soundVolume )
+{
+	QListIterator<QMediaPlayer *> itMusicBox( MusicBoxList );
+	QMediaPlayer * pPlayer=0;
+	
+	/* find first available inactive QMediaPlayer */
+	while( itMusicBox.hasNext() ) {
+		QMediaPlayer * pTestPlayer=itMusicBox.next();
+	
+		if ( pTestPlayer->state() != QMediaPlayer::PlayingState &&
+			 pTestPlayer->mediaStatus() != QMediaPlayer::LoadingMedia )
+		{
+			pPlayer=pTestPlayer;
+			break;
+		}
+	}
+
+	/* no available QMediaPlayer, create a new one */
+	if (!pPlayer) {
+		pPlayer = new QMediaPlayer(this);
```

Comment:
You've said your code allows an _unlimited_ number of  `QMediaPlayers` to be created but I think you must now check that you are able to do this - I think it would be wise to check that `pPlayer` is non-null and to error out if you can't.  This may need to extend the lua command playSound(...) to return a _nil_ + _error message_ in that event.

_Also, we might need to consider a mechanism to destroy 'spare' QMediaPlayers afterwards, or what happens if a user's script erroneously creates _lots_ of them in a (broken) loop?  It might be feasible (but more effort) to arrange for a one-shot timer to be coupled to a 'signal' that the the media has finished playing (or failed with an error) and to destroy the instance when the time-out has elapsed...?_